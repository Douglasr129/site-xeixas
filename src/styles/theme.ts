/* eslint-disable quotes */
export const theme = {
	colors: {
		primaryColor: '#0a1128',
		secondaryColor: '#dc143c',
		white: '#FFFFFF',
		mediumGray: '#DDDDDD',
		darkColor: '#2d2d2d',
	},
	font: {
		family: {
			default: "'Open Sans', sans-serif",
			secundary: "'Montserrat',sans-serif",
		},
		sizes: {
			xsmall: '0.8rem',
			small: '1.6rem',
			medium: '2.4rem',
			large: '3.2rem',
			xlarge: '4.0rem',
			xxlarge: '4.8rem',
			huge: '5.6rem',
			xhuge: '6.4rem',
		},
	},
	media: {
		lteMedium: '(max-width: 768px)',
	},
	spacing: {
		xsmall: '0.8rem',
		small: '1.6rem',
		medium: '2.4rem',
		large: '3.2rem',
		xlarge: '4.0rem',
		xxlarge: '4.8rem',
		huge: '5.6rem',
		xhuge: '6.4rem',
	},
};
