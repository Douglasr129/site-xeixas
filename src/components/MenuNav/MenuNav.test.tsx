import { screen, within } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import mock from './mock';
import { MenuNav } from '.';
import { theme } from '../../styles/theme';

describe('<Links />', () => {
	it('should render a link', () => {
		renderTheme(<MenuNav Listlinks={mock} />);
		expect(screen.getAllByRole('link')).toHaveLength(mock.length);
	});
	it('should render list of links', () => {
		renderTheme(<MenuNav Listlinks={mock} />);
		const list = screen.getByRole('list');
		const { getAllByRole } = within(list);
		const items = getAllByRole('listitem');
		expect(items.length).toBe(mock.length);
	});
	it('should render list of links', () => {
		renderTheme(<MenuNav />);
		const list = screen.getByRole('list');
		expect(list.childElementCount).toBe(0);
	});
	it('should render links in mobile', () => {
		renderTheme(<MenuNav Listlinks={mock} />);
		const element = screen.getByRole('list');
		expect(element).toHaveStyleRule({
			media: theme.medias.tablet,
		});
	});
	it('should render in dark mode', () => {
		renderTheme(<MenuNav Listlinks={mock} darkMode={true} />);
		const { container } = renderTheme(
			<MenuNav Listlinks={mock} darkMode={true} />,
		);
		expect(container.firstChild).toHaveStyleRule(
			'background',
			theme.colors.white,
		);
	});
	it('should render in dark mode', () => {
		const { container } = renderTheme(
			<MenuNav Listlinks={mock} darkMode={true} />,
		);
		expect(container.firstChild).toMatchSnapshot();
	});
});
