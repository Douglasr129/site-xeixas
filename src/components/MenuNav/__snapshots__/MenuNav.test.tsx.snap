// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Links /> should render in dark mode 1`] = `
.c0 {
  background: #FFFFFF;
  margin: 0;
}

.c1 {
  list-style: none;
  display: flex;
  flex-flow: row wrap;
  margin: 0;
  justify-content: center;
  padding: 0.8rem;
}

.c2 {
  padding: 1.6rem;
}

.c3 {
  padding: 0.8rem;
  margin: 0.8rem;
  text-decoration: none;
  color: #161421;
  text-align: center;
}

.c3:hover {
  border-bottom: solid 0.2rem #dc143c;
}

@media (max-width: 768px) {
  .c1 {
    flex-flow: column wrap;
    justify-content: space-evenly;
    align-content: center;
  }
}

<nav
  class="c0"
>
  <ul
    class="c1"
  >
    <li
      class="c2"
    >
      <a
        class="c3"
        href="#target1"
        target="_self"
      >
        Link 1
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="#target2"
        target="_self"
      >
        Link 2
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="#target3"
        target="_self"
      >
        Link 3
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="#target4"
        target="_self"
      >
        Link 4
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="#target5"
        target="_self"
      >
        Link 5
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="#target6"
        target="_self"
      >
        Link 6
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="#target7"
        target="_self"
      >
        Link 7
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="#target8"
        target="_self"
      >
        Link 8
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="#target9"
        target="_self"
      >
        Link 9
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="#target10"
        target="_self"
      >
        Link 10
      </a>
    </li>
  </ul>
</nav>
`;
